src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -83,7 +90,27 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService,
         }
 
         Participation participation = textSubmission.get().getParticipation();
-        participation  = participationService.findOneWithEagerResultsAndSubmissions(participation.getId());
+        participation = participationService.findOneWithEagerResultsAndSubmissions(participation.getId());
+        if (!participation.getResults().isEmpty()) {
+            User user = userService.getUser();
+            User assessor = participation.findLatestResult().getAssessor();
```

Comment:
What happens if there is a new submission (without result) and the latest result does not belong to this submission?

Reply:
There can be more submissions for every participation? And we want to assess them all?

Reply:
only the latest one

Reply:
I replaced with `participation.findLatestSubmission().getResult().getAssessor();`, and I eager loaded it